### PR TITLE
Add overworld glitch logic

### PR DESCRIPTION
**Pull request by compiling**
Adapted from qadan's work for door rando

### CODE REVIEW

File: BaseClasses.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def set_player_attr(attr, val):
             set_player_attr('can_access_trock_front', None)
             set_player_attr('can_access_trock_big_chest', None)
             set_player_attr('can_access_trock_middle', None)
-            set_player_attr('fix_fake_world', True)
+            set_player_attr('fix_fake_world', logic[player] not in ['owglitches', 'nologic'] or shuffle[player] in ['crossed', 'insanity', 'madness_legacy'])
```
Comment (aerinon) [inline BaseClasses.py:112]: ~~Crossed and Insanity ER allows Fake World now? I'm not sure about that.~~ oh, maybe I'm reading that wrong?

**Review by aerinon — COMMENTED** [review]
Initial review done, still need to address issues with original PR

File: BaseClasses.py (hunk 1 above)
Comment (compiling) [reply BaseClasses.py:112]: It's the other way.  Crossed and insanity fix fake worlds even with glitch logic.

**Comment by compiling** [comment]
For the issues with the original OWG PR:

* OWG connections are now only created with owglitches and nologic, so the dungeon generation will work as expected.
* Dungeon revive logic has a big list of rooms you can bunny revive in - currently incomplete but it errs on the side of requiring the moon pearl.  You are expected to bunny revive at the Skull circle of pots, since you get auto superbunny for falling down a hole.

**Review by aerinon — APPROVED** [review]

**Pull request merged**